downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md

```diff
@@ -140,6 +140,52 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
     3d|4f|6 b = 4; // Compilation error now, `4` is considered to be `int`.
     ```
 
+- A bug where an `anydata` value could be assigned to contexts requiring an `anydata & readonly` value has been fixed.
+
+    ```ballerina
+    public function main() {
+        anydata a = [];
+        anydata & readonly b = a; // Compilation error now.
+    }
+    ```
+  
+- A bug that allowed including a `readonly` `service` object inside a `readonly` non-`service` class has been fixed.
+
+    ```ballerina
+    public type A readonly & service object {
+        isolated remote function execute() returns int;
+    };
+  
+    readonly client class B {
+        *A; // Compilation error now.
+  
+        isolated remote function execute() returns int {
+            return 2;
+        }
+    }
+    ```
+
+- A bug where the runtime error messages were including the flattened representation of the union types instead of the name used in the type definition has been fixed.
```

Comment:
```suggestion
- A bug which caused the runtime error messages to include the flattened representation of a union type instead of the name used in the type definition has been fixed.
```